lsdcGui.py

```diff
@@ -4749,34 +4809,19 @@ def drawVector(self):
                                         self.centerMarker.y()+self.vectorEnd["graphicsitem"].y()+self.centerMarkerCharOffsetY, pen)
       self.vecLine.setFlag(QtWidgets.QGraphicsItem.ItemIsMovable, True)
 
-    def setVectorEndCB(self): #save sample x,y,z
-      if (self.vectorEnd != None):
-        self.scene.removeItem(self.vectorEnd["graphicsitem"])
-        try:
-          self.scene.removeItem(self.vecLine)
-        except AttributeError: # likely due to self.vecLine not being defined yet
-          pass
-        self.vectorEnd = None
-      self.vectorEnd = self.getVectorObject()
-
-      if self.vectorStart and self.vectorEnd:
-        self.drawVector()        
-
     def clearVectorCB(self):
-      if (self.vectorStart != None):
+      if self.vectorStart:
         self.scene.removeItem(self.vectorStart["graphicsitem"])
         self.vectorStart = None
-      if (self.vectorEnd != None):
+      if self.vectorEnd:
         self.scene.removeItem(self.vectorEnd["graphicsitem"])
         self.vectorEnd = None
         self.vecLenLabelOutput.setText("---")
         self.vecSpeedLabelOutput.setText("---")        
-      try:
-        if self.vecLine != None:
-          self.scene.removeItem(self.vecLine)
-      except AttributeError: # likely due to self.vecLine not defined yet
-        pass
-
+      if self.vecLine:
+        self.scene.removeItem(self.vecLine)
+        self.vecLine = None
+      
```

Comment:
do we know whether we are still getting these `AttributeError`s or not? (several areas of this commit) we may get more crashes of the GUI

Reply:
We shouldn't get attribute error for this specific variable as it is initialized to None: [here](https://github.com/NSLS-II/lsdc/blob/6dce1d9c5927b1b41394823e48526887458b8bd1/lsdcGui.py#L2203). So it should be safe to check its value anywhere in the app

Reply:
you're right, as long as it is defined, we shouldn't need that exception. resolving this.